packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -0,0 +1,831 @@
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username1', 'username2']);
+        });
+
+        it('removing configuration element renders the mention node as absolute link', async () => {
+            element.setMarkdown('<user:1>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            const renderedUserMention = element.lastElementChild as RichTextMentionUsers;
+            element.removeChild(renderedUserMention);
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual(['user:1']);
+        });
+
+        it('removing mapping element renders the mention node with user ID', async () => {
+            element.setMarkdown('<user:1>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            const renderedUserMention = element.lastElementChild as RichTextMentionUsers;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
+            renderedUserMention.removeChild(renderedMappingUser);
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['1']);
+        });
+
+        it('updating to invalid `pattern` in mention configuration converts it to absolute link', async () => {
+            element.setMarkdown('<user:1>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            (element.lastElementChild as RichTextMentionUsers).pattern = 'invalid';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual(['user:1']);
+        });
+
+        it('updating `display-name` in mapping mention should update the `mention-label` in view', async () => {
+            element.setMarkdown('<user:1>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+
+            const renderedUserMention = element.lastElementChild as RichTextMentionUsers;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
```

Comment:
We end up doing a lot of imperative dom lookups unnecessarily. Could you have appendUserMentionConfiguration return an object of what it created?
Something like:
```
{
    userMentionElement: RichTextMentionUsers,
    mappingElements: MappingUser[]
}
```
That way we can avoid all the `as` casting and the kind of confusing lastElementChild, firstElementChild lookups.

Reply:
Updated as per your suggestion.